### ISSUE TITLE
InputChip does not announce any action for a11y on iOS

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#58010]
b/154684205

'AssistiveChip' is only announced as a text in iOS though in Android it says 'double tap to activate'. We shall indicate that users could take action?

https://material.io/components/chips#action-chips

Minimal reproduction case:

```dart
    MaterialApp(
      title: 'Flutter Demo',
      theme: GoogleMaterial.theme(
        primarySwatch: GoogleColors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text('Home Page'),
        ),
        body: Center(
          child: InputChip(label: Text('test'), onPressed: () {}),
        ),
      ),
    );
```

**Comment by HansMuller** [comment]
AssistiveChip doesn't appear to be part of the Flutter API?

**Comment by mehmetf** [comment]
Ah, this is GM. Sorry!

I looked through the source code and it is mostly a theme around `InputChip`. Updated the example.

**Comment by nscobie** [comment]
I don't have a ton of context yet for how all the accessibility systems should be interacting, but after poking around a bit I'm thinking **this might technically be working as intended**. Further input from those with more experience would be appreciated.

Having a default announcement that interaction is possible for elements such as buttons is trivial, since their interaction is clear and unambiguous. Apple's UIKit exposes [UIAccessibilityTraits.button](https://developer.apple.com/documentation/uikit/uiaccessibility/uiaccessibilitytraits/1620194-button), which makes VoiceOver happily announce that an element is a _"button"_. This is how the standard implementation for Material components on iOS [specifies](https://github.com/material-components/material-components-ios/blob/develop/components/Buttons/src/MDCButton.m#L588) that buttons should be announced, and Flutter does the same when `button: true` is set on a `Semantics` node.

**However, `InputChip`s don't always behave just like buttons. It looks like the standard implementation for Material components on iOS [does not specify](https://github.com/material-components/material-components-ios/blob/develop/components/Chips/src/MDCChipField.m#L134) any [UIAccessibilityTraits](https://developer.apple.com/documentation/uikit/uiaccessibility/uiaccessibilitytraits) for an `InputChip`, and I'm guessing this is because the semantics for interacting with an `InputChip` are unique to the use case.**

Unfortunately, unlike Android's TalkBack, it seems like VoiceOver does not announce a generic "double tap to activate" message on all elements which can be activated. Therefore, my hunch is that one should **always use a `Semantic` node to describe the functionality of a `*Chip`.**

If this is indeed the case, I would be happy to help update the docs to recommend doing so. That being said, I might be missing something simple here.

/cc @gaaclarke

**Comment by gaaclarke** [comment]
@nscobie your linked code is for the MDCChipField which isn't analogous to the InputChip.  You can see here that the chip should be reporting itself as a button: https://github.com/material-components/material-components-ios/blob/e1aee8556d04a64cef5149d51897f44711a5b077/components/Chips/tests/unit/ChipsTests.m#L100

We should make sure that `button: true` is happening so UIAccessibilityTraitButton is set.  That makes sense to me.  I worry that it will just announce "button" and not "double tap to activate".  The button announcement is better than nothing as a first step though.

**Comment by chunhtai** [comment]
IOS cannot customize an element to have double tap to xx, only android can do that. @gaaclarke do you think we should apply the button to all chip classes?

**Comment by nscobie** [comment]
@gaaclarke Ah, you're absolutely right. I was confused by the [Chip docs](https://github.com/material-components/material-components-ios/tree/develop/components/Chips#input-chips) claiming:

> We currently provide an implementation of Input Chips called `MDCChipField`.

But after rereading the code I [linked](https://github.com/material-components/material-components-ios/blob/develop/components/Chips/src/MDCChipField.m#L134) I see that's more of a container/wrapper for `MDCChipView`, and `UIAccessibilityTraitNone` is just being applied to a text field. :|

After learning more [about chips](https://material.io/components/chips), it makes a lot more sense that they would generally be considered as buttons. However, it still feels a little odd to me to always mark `InputChip`s in particular as buttons, since interacting with them seems to usually be more of an "edit" action than an "activate this thing" action. But I guess `button: true` would still be a sensible default which users could override with `Semantics` and `ExcludeSemantics` nodes on a case by case basis?

**Comment by nscobie** [comment]
This was reverted due to breaking internal tests. Reopening this issue until we can reland the PR.
